roseau/load_flow/models/lines/parameters.py

```diff
@@ -186,14 +233,68 @@ def insulator_type(self) -> InsulatorType | None:
+        """The cross-section areas of the cable (in mm²). Informative only, it has no impact on the load flow."""
+        return None if self._sections is None else Q_(self._sections, "mm**2")
 
-    @max_current.setter
+    @max_currents.setter
     @ureg_wraps(None, (None, "A"))
-    def max_current(self, value: float | Q_[float] | None) -> None:
-        self._max_current = value
+    def max_currents(self, value: FloatScalarOrArrayLike1D | None) -> None:
+        value_isna = pd.isna(value)
+        if not np.isscalar(value_isna):
+            value_isna = np.array(value_isna).all()
+
+        if value_isna:
+            values = None
+        else:
+            if np.isscalar(value):
+                if value < 0:
+                    msg = f"Maximum currents must be non-negative: {value} A was provided."
+                    logger.error(msg)
+                    raise RoseauLoadFlowException(msg=msg, code=RoseauLoadFlowExceptionCode.BAD_MAX_CURRENTS_VALUE)
+                value = [value for _ in range(self._size)]
+
+            values = np.array(value, dtype=np.float64)
+            if len(values) != self._size:
+                msg = f"Incorrect number of maximum currents: {len(values)} instead of {self._size}"
+                logger.error(msg)
+                raise RoseauLoadFlowException(msg=msg, code=RoseauLoadFlowExceptionCode.BAD_MAX_CURRENTS_SIZE)
+            if (values < 0).any():
+                msg = f"Maximum currents must be non-negative: [{', '.join(f'{x}' for x in values)}] A was provided."
+                logger.error(msg)
+                raise RoseauLoadFlowException(msg=msg, code=RoseauLoadFlowExceptionCode.BAD_MAX_CURRENTS_VALUE)
+        self._max_currents = values
+
+    @sections.setter
+    @ureg_wraps(None, (None, "mm**2"))
+    def sections(self, value: FloatScalarOrArrayLike1D | None) -> None:
+        value_isna = pd.isna(value)
+        if np.isscalar(value_isna):
+            if value_isna:
+                values = None
+            elif value < 0:
+                msg = f"Sections must be non-negative: {value} mm² was provided."
+                logger.error(msg)
+                raise RoseauLoadFlowException(msg=msg, code=RoseauLoadFlowExceptionCode.BAD_SECTIONS_VALUE)
+            else:
+                values = np.array([value for _ in range(self._size)], dtype=np.float64)
+        else:
+            if np.any(value_isna):
+                msg = f"Sections can't be null: [{', '.join(f'{x}' for x in value)}] mm² was provided."
+                logger.error(msg)
+                raise RoseauLoadFlowException(msg=msg, code=RoseauLoadFlowExceptionCode.BAD_SECTIONS_VALUE)
+            if len(value) != self._size:
+                msg = f"Incorrect number of sections: {len(value)} instead of {self._size}"
+                logger.error(msg)
+                raise RoseauLoadFlowException(msg=msg, code=RoseauLoadFlowExceptionCode.BAD_SECTIONS_SIZE)
+
+            values = np.array(value, dtype=np.float64)
```

Comment:
Move this line to the beginning of the `else` block so that we are sure we are manipulating numpy arrays when doing the checks (like when calling `len(value)` above or when calling `value.tolist()` as suggested in the comment above.